capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -88,45 +74,40 @@ public PostService build() {
   private PostService(Builder builder) {
     this.blobstore = builder.blobstore;
     this.datastore = builder.datastore;
-    this.imagesService = builder.imagesService;
     this.clock = builder.clock;
     postSorters = new HashMap<>();
     initializeSorters();
   }
 
-  /** Returns the uploaded images' url if an image was uploaded. Otherwise, returns null. */
-  public String uploadImage(HttpServletRequest request) {
+  /**
+   * Returns the uploaded files' blob key string representation if a file was uploaded. Otherwise,
+   * returns null.
+   */
+  public String uploadFile(HttpServletRequest request) {
```

Comment:
I'd consider using Java's Optional if you want to return null for objects.

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
I am storing the string inside of datastore and it looks like Optional is not supported by datastore. I could use the toString method when storing the value but I feel like that's like storing the string but with extra steps

Reply:
Optional is not used for storage, it is just used to ease passing our objects in your code when handling with any logic that might return null. It makes it so that it forces you to check for null when passed around so that you can avoid NullPointerExceptions.

Reply:
I'm still a bit confused. I attempted it but I'm not sure if I did it entirely right.